includes/rest-api/class-amp-rest-fonts-controller.php

```diff
@@ -0,0 +1,242 @@
+<?php
+/**
+ * Class AMP_REST_Fonts_Controller
+ *
+ * @package AMP
+ */
+
+/**
+ * Basic font api for the AMP stories editor.
+ *
+ * Class AMP_REST_Fonts_Controller
+ */
+class AMP_REST_Fonts_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'amp/v1';
+		$this->rest_base = 'fonts';
+	}
+
+	/**
+	 * Registers routes for amp fonts.
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			[
+				[
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => [ $this, 'get_items' ],
+					'permission_callback' => [ $this, 'get_items_permissions_check' ],
+					'args'                => $this->get_collection_params(),
+				],
+				'schema' => [ $this, 'get_public_item_schema' ],
+			]
+		);
+	}
+
+	/**
+	 * Gets a collection of fonts.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 *
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
+		$fonts       = AMP_Fonts::get_fonts();
+		$total_posts = count( $fonts );
```

Comment:
I suppose this should be `$total_fonts`?

Is the check for that necessary for fonts? Could you explain the logic behind allowing requesting the fonts using `per_page` and `page` params, in which cases are these being used in the editor currently? Is this for the future, for loading the fonts in batches in the dropdown?

Reply:
Correct, this variable should be changed. I have written the API should it can be changed to be paginated easily later, if we want to lazy load in fonts.

Reply:
Changed in https://github.com/ampproject/amp-wp/pull/3998/commits/b6bffec9e1b0972aff684ead1b8ed9062cd40897